x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/support/SecurityMigrationTaskParams.java

```diff
@@ -27,27 +27,33 @@ public class SecurityMigrationTaskParams implements PersistentTaskParams {
 
     private final int migrationVersion;
 
+    private final boolean migrationNeeded;
+
     public static final ConstructingObjectParser<SecurityMigrationTaskParams, Void> PARSER = new ConstructingObjectParser<>(
         TASK_NAME,
         true,
-        (arr) -> new SecurityMigrationTaskParams((int) arr[0])
+        (arr) -> new SecurityMigrationTaskParams((int) arr[0], (boolean) arr[1])
     );
 
     static {
         PARSER.declareInt(constructorArg(), new ParseField("migration_version"));
+        PARSER.declareBoolean(constructorArg(), new ParseField("migration_needed"));
```

Comment:
I think this needs to be an `optionalConstructorArg`, work with `Boolean` (and default it to `true`).
This is a thing that goes into the cluster state so it could be that we deserialize and serialize from different versions.

Reply:
Good catch! I've updated this to be an optional + added a null check. Thanks!